src/Data/List/Relation/Unary/All/Properties.agda

```diff
@@ -335,6 +337,20 @@ module _ {P : Pred A p} {Q : Pred A q} where
 
 ------------------------------------------------------------------------
 -- Introduction (⁺) and elimination (⁻) rules for list operations
+------------------------------------------------------------------------
+-- head, last
+
+module _ {P : A → Set p} where
+
+  head⁺ : ∀ {xs} → All P xs → Maybe.All P (head xs)
+  head⁺ []       = nothing
+  head⁺ (px ∷ _) = just px
+
+  last⁺ : ∀ {xs} → All P xs → Maybe.All P (last xs)
+  last⁺ []             = nothing
+  last⁺ (px ∷ [])      = just px
+  last⁺ (px ∷ py ∷ xs) = last⁺ (py ∷ xs)
```

Comment:
same pattern here as @gallais pointed out above.

Reply:
Here we will not be able to avoid the split because we need `last`
to reduce in the type. The best we can hope for is something like
`last⁺ (px ∷ pxs@(_ ∷ _)) = last⁺ pxs` which is not necessarily
more readable.

Reply:
Thanks - makes sense.